### PR TITLE
Fixes #1221 : Added HasElementXBeforeElementY Classifier Rule [Blocked: #1092]

### LINKED ISSUE #1221: Add classifiers hasElementXBeforeY
Add new classifier rule `hasElementXBeforeY`

**Target PR date**: 23 May 2020
**Target completion date**: 25 May 2020

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
## Explanation

Fixes #1221 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt
Comment (BenHenning) [inline domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt]: Is this comment correct? It seems like the types of the two parameters are the same, so is the ordering actually relevant to the test being done here?

File: domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt
Comment (BenHenning) [inline domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt]: Nit here and elsewhere: ‘incorrectly’ should be spelled as one word.

File: domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt
Comment (BenHenning) [inline domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt]: Nit here and elsewhere in the test: space before the ‘,’

Be sure to double check the formatting beforehand. I suggest running ktlint to make sure these types of things are caught early.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @aggarwalpulkit596! Two additional things:
1. Please update the PR title to include the issue it’s fixing
2. Please add a note in the title indicating which PR this one is blocked on to be submitted first (since it’s not based on develop). Ditto for all of your other PRs.

File: domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt
Comment (aggarwalpulkit596) [reply domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt]: i did run ktlint might have been a mistake from my end while checking things.

File: domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt
Comment (aggarwalpulkit596) [reply domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt]: yes will remove

Comment (aggarwalpulkit596) [reply domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt]: done

File: domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt
Comment (aggarwalpulkit596) [reply domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt]: done

File: domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt
Comment (aggarwalpulkit596) [reply domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYRuleClassifierProviderTest.kt]: fixed

File: domain/src/main/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYClassifierProvider.kt
Comment (vinitamurthi) [inline domain/src/main/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYClassifierProvider.kt:11]: These lines seem to be too long. Here and elsewhere

**Review by vinitamurthi — APPROVED** [review]
Syntactically looks good, some issues about line being too long. I will leave it to @BenHenning to check the semantic correctness of the classifier

**Review by BenHenning — APPROVED** [review]
Thanks @aggarwalpulkit596! This LGTM!

File: domain/src/main/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYClassifierProvider.kt
Comment (aggarwalpulkit596) [reply domain/src/main/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputHasElementXBeforeElementYClassifierProvider.kt:11]: it should be max 100 lines?

**Pull request merged**
